nixos/modules/services/networking/mullvad-vpn.nix

```diff
@@ -65,6 +65,9 @@ with lib;
         pkgs.iproute2
         # Needed for ping
         "/run/wrappers"
+        # See https://github.com/NixOS/nixpkgs/issues/262681
+        (mkIf config.networking.resolvconf.enable
+          config.networking.resolvconf.package)
```

Comment:
Typically, `mkIf` is used as a part of of `mkMerge`. When constructing lists, `lib.optional`/`lib.optionals` is used instead.

```patch
-        (mkIf config.networking.resolvconf.enable
-          config.networking.resolvconf.package)
-      ];
+      ] ++ (lib.optional config.networking.resolvconf.enable
+        config.networking.resolvconf.package);
```

Reply:
thanks, updated